### PR TITLE
Follow the latest NumPy/SciPy docs style

### PR DESCRIPTION
**Pull request by kmaehashi**
Tried to follow the latest NumPy API docs style.

* Removed `:nosignatures:`. The new theme is wide enough to show a full signature for methods.
* Use `currentmodule` to omit module name from the list.

Also added a link to the original NumPy API reference so that users can quickly compare it with NumPy ones.

I only updated creation and manipulation routines and would like to hear what you fell.

### CODE REVIEW

**Comment by kmaehashi** [comment]
This is ready for review. This PR also includes:

* Add `cupy.vectorize` to docs
* Fix several APIs left undocumented (e.g., `cupy.conjugate`, which are indicated as a dead link in https://docs.cupy.dev/en/v9.0.0rc1/reference/comparison.html)
* Remove `cupy.bitwise_not` from the list, as it is undocumented in NumPy
* Bump sphinx/pydata_sphinx_theme/numpy veresion

**Comment by kmaehashi** [comment]
pfnCI, test this please.

File: docs/source/reference/linalg.rst [hunk 1]
```diff
@@ -1,105 +1,71 @@
...
+.. currentmodule:: cupy
 
 Matrix and vector products
 --------------------------
 
 .. autosummary::
    :toctree: generated/
-   :nosignatures:
-   
-   cupy.cross
```
Comment (kmaehashi) [inline docs/source/reference/linalg.rst:13]: note: `cross` moved to `math.rst`.

File: docs/source/reference/logic.rst [hunk 2]
```diff
@@ -1,74 +1,69 @@
...
 -------------------
 
 .. autosummary::
    :toctree: generated/
-   :nosignatures:
 
-   cupy.all
-   cupy.any
-   cupy.in1d
-   cupy.isin
```
Comment (kmaehashi) [inline docs/source/reference/logic.rst:16]: note: These are moved to set.rst (new page)

File: docs/source/reference/math.rst [hunk 3]
```diff
@@ -1,181 +1,180 @@
...
-   cupy.minimum
-   cupy.fmax
-   cupy.fmin
-   cupy.nan_to_num
-   cupy.interp
-   cupy.bartlett
-   cupy.blackman
-   cupy.hamming
-   cupy.hanning
-   cupy.kaiser
```
Comment (kmaehashi) [inline docs/source/reference/math.rst:181]: note: These are moved to window.rst (new page)

**Review by emcastillo — APPROVED** [review]
LGTM

**Comment by kmaehashi** [comment]
I'll take care of backport

**Pull request merged**